birdhouse/scripts/create-magpie-users

```diff
@@ -69,6 +69,9 @@ THIS_FILE="`realpath "$0"`"
 THIS_DIR="`dirname "$THIS_FILE"`"
 COMPOSE_DIR="`dirname "$THIS_DIR"`"
 
+# must retrieve MAGPIE_VERSION
+. "${COMPOSE_DIR}/default.env"
```

Comment:
Surround this with `if [ -f "${COMPOSE_DIR}/default.env" ]`, like `env.local`, so this script retain it's standalone feature.

Reply:
Shouldn't this file be enforced to make sure the variables exist?
It is this kind of situation that justifies why I prefer to add the "required" variables (eg Tag version of docker images her) in `VARS` as in other PRs, although they *should* always exist from loading `defaults.env`. 
Not having those variables would break the whole server deployment and will not necessarily be easy to debug opposed to a clear message "undefined variable X".

Reply:
It's very convenient to be able to run this script in standalone mode, meaning running the script outside of a proper checkout of birdhouse-deploy repo and without `env.local` matching the intended VM.  I actually use it directly on my machine to hit several of my VM, instead of having to ssh to each of my VM when I need it.  Of course, I override those env vars for the intended VM.

From a proper checkout, everything should resolve.  And even with the good `default.env`, `env.local` could still accidentally override it to an invalid value.  So at this stage, it's user error that is intentional, maybe to test an alternate version of Magpie.

So my point is even if we force sourcing the `default.env`, we still have no warranty that the var will have the same value as `default.env` and we lose the "standalone" feature.  So I'd rather keep the standalone feature.

Reply:
I'm not against skipping sourcing `default.env` per se. 
I'm questioning the original reason why we were not validating that variables where defined because it was *assumed* that they exists since this file is always force-sourced. If this is not the case anymore because indeed they can still be defined through `env.local` instead, I believe it is in everyone's interest to have a validation check that they were indeed defined that way.

Reply:
Yes, indeed, you can copy/paste the var validation code from `pavics-compose.sh` https://github.com/bird-house/birdhouse-deploy/blob/edd8ab40f783b6b724b88c1abd5c2ac1d721f8dc/birdhouse/pavics-compose.sh#L87-L95 for the 4 vars: URL, USER, PASS, IMAGE.

To be strict, there are a bunch of other scripts that source `default.env` and `env.local` that would benefit from the same var validation code as well.  Can be done in another  PR of course.

Reply:
Ok. Let's move it for another PR.

Reply:
Agreed, let's keep the standalone feature now and add var validation at the same time as all the other scripts in the same situation.

Reply:
For reference #172

Reply:
You forgot to restore standalone feature (surround the sourcing of `default.env` with an `if` statement).